tests/unit/misc/test_autoupdate.py

```diff
@@ -0,0 +1,104 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>:
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.misc.autoupdate"""
+
+from PyQt5.QtTest import QSignalSpy
+
+from qutebrowser.misc import autoupdate, httpclient
+
+
+class HTTPGetStub(httpclient.HTTPClient):
+    """A stub class for HTTPClient.
+
+        Attributes:
+        _success: Wether get() will emit a success signal.
+    """
+
+    def __init__(self, success=True, json=None):
+        self._success = success
+        if json:
+            self._json = json
+        else:
+            self._json = '{"info": {"version": "test"}}'
+
+        super(HTTPGetStub, self).__init__()
```

Comment:
You can/should use the Python 3 syntax here: `super().__init__()` - also I'd suggest moving the superclass init to the top of `__init__` so it gets executed first. It doesn't make a difference here, but it's the usual thing to do (tm) :wink: